Move the stroke parsing to the protocol file

whot opened the pull request:
Biggest advantage here: we can test all this.

I was hoping that the code would become a bit more understandable in the process but, well, it's not that much different to what we had anyway. The specs are mostly wrong or missing crucial details, so this is largely implemented on the RE-d protocol with a few updates from the specs.

Notable: in the stroke header we can now extract the pen id, pen type and the new layer flag (all unused though). We can deal with missing points though I haven't seen those yet anywhere.

For the tests: there are a few hand-composed ones but more usefully: on every test run it will run through the local `.yaml` raw files and re-parse those. So the more of those are lying around (and they accmulate with usage unless you explicitly clear them out), the better the test exposure gets.

Filing as draft because I want this to be tested on the Slate before we consider merging it. It seems to be happy with my intuos and spark so far.

whot committed:
wacom: group the list2hex output

Group by size 8 by default.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot committed:
protocol: add stroke parsing

This is a slightly different model as the messages, primarily because it's not
quite as model-specific. So there's only one parse function and it can handle
both file types that we currently support (intuos pro and the spark/slate
bits).

All wrapped into their own classes to make future extensions a bit easier.
It's not a 1:1 implementation of the tuhi/wacom.py bits either because we now
know about a few extra bits like the flags in stroke headers.

Most importantly though, this can be easily tested now, with one test case
searching for raw logs in $XDG_DATA_HOME/tuhi/raw and parsing all of those. So
the more files are sitting in there (i.e. the more the tablet is used), the
better the test suite becomes.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot committed:
Add a utility function module

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot committed:
tools: add an svg generation tool

This takes the raw log YAML file tuhi saves for each interaction, extracts the
pen data and then creates an SVG from that. Good for testing whether the
stroke parsing is sane.

Where --all is given, it will search through the directory that tuhi uses to
store the raw logs.

Two fixmes: the pointsize is hardcoded to 5 (intuos pro) because there's no
easy way here to get the point size. Same for the pressure range.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot committed:
wacom: switch to the new protocol stroke parsers

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

bentiss commented:
Hmm, I guess there is something wrong here. In the following, I wrote `test`, and only the first `t` gets recognized:

```
09:31:43 DEBUG: tuhi: E0:61:C6:BF:14:4E: UUID 3503516e7e74 protocol: slate
09:31:43 INFO: tuhi.ble.E0:61:C6:BF:14:4E: Connecting
09:31:44 DEBUG: tuhi: E0:61:C6:BF:14:4E: UUID 3503516e7e74 protocol: slate
09:31:44 INFO: tuhi.ble.E0:61:C6:BF:14:4E: Connecting
09:31:44 DEBUG: tuhi.ble.E0:61:C6:BF:14:4E: Already connecting
09:31:44 INFO: tuhi.ble.E0:61:C6:BF:14:4E: Connection established
09:31:45 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX SYSEVENT <-- b'f1024200'
09:31:45 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX SYSEVENT <-- b'f1024200'
09:31:46 DEBUG: tuhi: E0:61:C6:BF:14:4E: connected for DeviceMode.LISTEN
09:31:46 DEBUG: tuhi.wacom: E0:61:C6:BF:14:4E: starting
09:31:46 DEBUG: tuhi.fw: command: check connection
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> e6 / 06 / 35 03 51 6e 7e 74
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:46 DEBUG: tuhi.fw: command: set time
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b6 / 06 / 19 08 20 07 31 46
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:46 DEBUG: tuhi.fw: command: read battery info
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b9 / 01 / 00
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- ba 02 42 00
09:31:46 INFO: tuhi.wacom: device battery: 66% (discharging)
09:31:46 DEBUG: tuhi.fw: command: get dimensions
09:31:46 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> ea / 02 / 03 00
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- eb 06 03 00 60 54 00 00
09:31:47 DEBUG: tuhi.fw: command: get dimensions
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> ea / 02 / 04 00
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- eb 06 04 00 d0 39 00 00
09:31:47 INFO: tuhi.wacom: dimensions: 21600x14800, point size 10µm
09:31:47 DEBUG: tuhi.fw: command: get firmware
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b7 / 01 / 00
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b8 09 00 0b 01 06 00 08    00 05 0a
09:31:47 DEBUG: tuhi.fw: command: get firmware
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b7 / 01 / 01
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b8 09 01 0c 01 06 00 09    02 03 0a
09:31:47 INFO: tuhi.wacom: firmware is b160805a-c160923a
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> ec / 06 / 06 00 00 00 00 00
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:47 DEBUG: tuhi.fw: command: set mode
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b1 / 01 / 01
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:47 DEBUG: tuhi.fw: command: set mode
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> b1 / 01 / 01
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:47 DEBUG: tuhi.fw: command: check for data
09:31:47 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> c1 / 01 / 00
09:31:48 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- c2 02 01 00
09:31:48 DEBUG: tuhi.wacom: Drawings available: 1
09:31:48 DEBUG: tuhi.fw: command: fetch data
09:31:48 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> cc / 01 / 00
09:31:48 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- cf 0a 70 03 00 00 19 08    20 07 31 42
09:31:48 INFO: tuhi.wacom: receiving 880 bytes drawn on UTC 190820073142
09:31:48 DEBUG: tuhi.fw: command: start reading
09:31:48 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> c3 / 01 / 00
09:31:48 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- c8 01 be
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 62 38 62 74 3f ff ee ee    ee d9 b5 ff ff ff 08 2c    b5 17 92 02
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- c0 56 03 e8 f5 06 2a 04    e8 0b fa df 04 e8 f5 04    86 05 e8 0b
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fc 21 06 e8 f7 02 a9 06    a8 09 fe 69 80 e2 80 eb    80 f5 80 f8
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 80 f7 80 ea 80 fc 00 00    00 00 28 0b 01 20 fe 28    02 01 28 03
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- ff 28 03 ff 08 02 08 02    08 02 08 01 08 01 20 ff    28 01 01 28
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- ff fe 28 fe 01 28 ff 01    28 ff ff 28 ff 02 08 fe    08 fd 28 fc
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 01 08 fd 08 f6 08 09 08    f7 00 00 00 00 00 28 f6    05 28 0a fb
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 28 ec 02 28 07 ff 28 fb    01 08 ff 28 ff ff 08 02    08 02 08 02
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 00 28 01 01 20 fd 28 01    01 28 03 01 20 fe 20 01    08 09 28 f1
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fe 28 0f 02 28 f7 fd 28    09 03 00 28 f6 fb 28 0a    05 00 28 ff
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f2 28 ff 02 28 fc fb 28    fe fc 28 f9 fd 28 ff fd    08 fe 28 03
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 06 28 07 02 28 05 08 28    03 0b 28 0b f8 20 05 28    f5 03 28 0e
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f6 28 f2 0a 28 f6 f1 28    0a 0f 28 f3 f8 28 0d 08    00 00 00 00
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 00 00 00 80 fe 80 80 80    91 c0 14 05 e8 0c 11 99    03 e8 fe f9
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 3c 01 fc ff ff ff ff ff    ff ff ff ee ee ee 6d b6    00 00 ff ff
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- ff 20 2d a2 16 08 02 80    76 80 17 80 fc a8 02 0a    f5 a8 fe f6
09:31:49 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f6 80 f6 80 fa 80 fa 80    f3 80 f5 80 f2 80 f2 80    f5 80 f9 80
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fd 80 fd 80 ff 80 ff 80    fe a8 fa f5 03 28 03 02    a8 ff fd fd
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- a8 fd ff fd 80 fb a0 02    01 a8 fe 02 01 88 01 ff    a8 ff 01 03
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- a8 09 08 fb a8 f3 f6 02    a8 0d 0a 02 80 ff 88 f5    01 88 0b fe
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 80 fd 00 a8 f6 08 fe a8    0a f8 02 a8 ff 09 fd 28    01 f7 a8 04
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 0a ff a8 fc f6 ff a8 0a    12 01 a8 ff fb 01 a0 01    ff 88 04 ff
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- a8 01 fe 01 a8 ff ff fe    a0 fd 02 20 ff 20 fe 08    01 20 fd 28
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fe fd a8 fe fe 01 a8 f6    03 fe a8 10 f5 01 a8 f0    0b ff a8 0c
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f0 03 a8 f7 05 ff a8 ff    fc ff 88 fe 04 a8 ff ff    ff a8 fd fe
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 02 a8 fe ff 02 a8 fd fd    ff a8 fc 01 01 08 ff a8    ff 02 fc a8
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fe 02 fd 00 20 02 28 ff    03 20 02 20 04 28 ff 01    28 02 02 28
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 03 01 28 03 01 28 0b 01    28 f3 04 28 0d fc 00 28    f4 0b 28 0c
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f5 00 28 fd 0b 28 03 f5    00 00 00 00 00 00 28 09    01 28 f7 ff
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 28 07 f7 28 f9 09 28 0c    ef 28 fa 06 28 01 fe 28    01 fd 28 01
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- fc 28 01 ff 00 28 01 01    28 ff 05 28 01 03 28 ff    03 20 03 28
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 01 02 28 01 02 28 ff 03    08 ff 28 01 03 28 f5 fc    28 0d 0c 28
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f8 fe a8 fd 01 d9 a8 fc    ff e4 88 fe d0 a8 fc ff    ba a8 fe fd
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- b2 a8 fb 01 c5 a8 fc fc    17 e8 fc ff fa 00 fc ff    ff ff ff ff
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- ff ff ff ee ee ee fa b6    00 00 ff ff ff 70 2b d5    14 55 02 c0
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 11 03 e8 ed 0b e9 03 e8    05 f5 a7 04 c8 ff 5d 05    e0 ff 0a 06
09:31:50 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- c8 02 a1 06 e8 02 ff 28    07 a8 0b 02 5b a8 f3 fb    e1 a8 0d 05
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f4 80 e0 80 f0 00 00 00    00 28 0d fc 28 fc 03 28    02 fe 28 06
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- ff 28 05 ff 28 05 fe 28    05 01 28 03 01 08 01 28    02 01 20 ff
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 28 fe ff 28 ff ff 28 fc    fe 28 fa 02 28 fc 01 28    fb ff 28 fc
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 03 08 ff 28 f5 04 28 0d    fa 28 f3 06 00 00 00 00    00 28 f3 fe
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 28 04 01 28 fe fe 08 fe    20 fe 28 ff fd 28 ff fe    20 fd 08 ff
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- 28 02 fe 20 fe 28 ff fc    20 fc 28 01 fc 28 fe f8    28 02 f8 20
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX PEN <-- f5 a0 f9 88 e8 08 fb e9    05 c8 02 17 03 fc ff ff    ff ff ff ff
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- c8 05 ed 75 0b 2f 2c
09:31:51 INFO: tuhi.wacom: received b'623862743fffeeeeeed9b5ffffff082cb5179202c05603e8f5062a04e80bfadf04e8f5048605e80bfc2106e8f702a906a809fe6980e280eb80f580f880f780ea80fc00000000280b0120fe2802012803ff2803ff0802080208020801080120ff28010128fffe28fe0128ff0128ffff28ff0208fe08fd28fc0108fd08f6080908f7000000000028f605280afb28ec022807ff28fb0108ff28ffff0802080208020028010120fd28010128030120fe2001080928f1fe280f0228f7fd2809030028f6fb280a050028fff228ff0228fcfb28fefc28f9fd28fffd08fe28030628070228050828030b280bf8200528f503280ef628f20a28f6f1280a0f28f3f8280d080000000000000080fe80808091c01405e80c119903e8fef93c01fcffffffffffffffffeeeeee6db60000ffffff202da21608028076801780fca8020af5a8fef6f680f680fa80fa80f380f580f280f280f580f980fd80fd80ff80ff80fea8faf503280302a8fffdfda8fdfffd80fba00201a8fe02018801ffa8ff0103a80908fba8f3f602a80d0a0280ff88f501880bfe80fd00a8f608fea80af802a8ff09fd2801f7a8040affa8fcf6ffa80a1201a8fffb01a001ff8804ffa801fe01a8fffffea0fd0220ff20fe080120fd28fefda8fefe01a8f603fea810f501a8f00bffa80cf003a8f705ffa8fffcff88fe04a8ffffffa8fdfe02a8feff02a8fdfdffa8fc010108ffa8ff02fca8fe02fd00200228ff032002200428ff01280202280301280301280b0128f304280dfc0028f40b280cf50028fd0b2803f500000000000028090128f7ff2807f728f909280cef28fa062801fe2801fd2801fc2801ff0028010128ff0528010328ff03200328010228010228ff0308ff28010328f5fc280d0c28f8fea8fd01d9a8fcffe488fed0a8fcffbaa8fefdb2a8fb01c5a8fcfc17e8fcfffa00fcffffffffffffffffeeeeeefab60000ffffff702bd5145502c01103e8ed0be903e805f5a704c8ff5d05e0ff0a06c802a106e802ff2807a80b025ba8f3fbe1a80d05f480e080f000000000280dfc28fc032802fe2806ff2805ff2805fe280501280301080128020120ff28feff28ffff28fcfe28fa0228fc0128fbff28fc0308ff28f504280dfa28f306000000000028f3fe28040128fefe08fe20fe28fffd28fffe20fd08ff2802fe20fe28fffc20fc2801fc28fef82802f820f5a0f988e808fbe905c8021703fcffffffffffff'
09:31:51 DEBUG: tuhi.protocol: FileHeader: time: 190820073151, stroke count: None
09:31:51 DEBUG: tuhi.protocol: next data packet UNKNOWN: 3f ff ee ee ee d9 b5 ff    ff ff 08 2c b5 17 92 02 …
09:31:51 DEBUG: tuhi.protocol: Unknown packet: <tuhi.protocol.StrokePacketUnknown object at 0x7f4d703e6ef0>
09:31:51 DEBUG: tuhi.protocol: offset 0: <tuhi.protocol.StrokePacketUnknown object at 0x7f4d703e6ef0>
09:31:51 DEBUG: tuhi.protocol: next data packet POINT: ff ff ff 08 2c b5 17 92    02 c0 56 03 e8 f5 06 2a …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11272, y=6069, p=658)
09:31:51 DEBUG: tuhi.protocol: offset 7: StrokePoint: 11272/6069 pressure: 658
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: c0 56 03 e8 f5 06 2a 04    e8 0b fa df 04 e8 f5 04 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11272, y=6069, p=854)
09:31:51 DEBUG: tuhi.protocol: offset 16: StrokeDelta:      /      pressure:   854
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 f5 06 2a 04 e8 0b fa    df 04 e8 f5 04 86 05 e8 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11261, y=6075, p=1066)
09:31:51 DEBUG: tuhi.protocol: offset 19: StrokeDelta:   -11/   +6 pressure:  1066
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 0b fa df 04 e8 f5 04    86 05 e8 0b fc 21 06 e8 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11261, y=6075, p=1247)
09:31:51 DEBUG: tuhi.protocol: offset 24: StrokeDelta:   +11/   -6 pressure:  1247
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 f5 04 86 05 e8 0b fc    21 06 e8 f7 02 a9 06 a8 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11250, y=6079, p=1414)
09:31:51 DEBUG: tuhi.protocol: offset 29: StrokeDelta:   -11/   +4 pressure:  1414
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 0b fc 21 06 e8 f7 02    a9 06 a8 09 fe 69 80 e2 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11250, y=6079, p=1569)
09:31:51 DEBUG: tuhi.protocol: offset 34: StrokeDelta:   +11/   -4 pressure:  1569
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 f7 02 a9 06 a8 09 fe    69 80 e2 80 eb 80 f5 80 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=1705)
09:31:51 DEBUG: tuhi.protocol: offset 39: StrokeDelta:    -9/   +2 pressure:  1705
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: a8 09 fe 69 80 e2 80 eb    80 f5 80 f8 80 f7 80 ea …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=1810)
09:31:51 DEBUG: tuhi.protocol: offset 44: StrokeDelta:    +9/   -2 pressure:  +105
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 e2 80 eb 80 f5 80 f8    80 f7 80 ea 80 fc 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=1885)
09:31:51 DEBUG: tuhi.protocol: offset 48: StrokeDelta:      /      pressure:   -30
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 eb 80 f5 80 f8 80 f7    80 ea 80 fc 00 00 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=1939)
09:31:51 DEBUG: tuhi.protocol: offset 50: StrokeDelta:      /      pressure:   -21
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 f5 80 f8 80 f7 80 ea    80 fc 00 00 00 00 28 0b …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=1982)
09:31:51 DEBUG: tuhi.protocol: offset 52: StrokeDelta:      /      pressure:   -11
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 f8 80 f7 80 ea 80 fc    00 00 00 00 28 0b 01 20 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2017)
09:31:51 DEBUG: tuhi.protocol: offset 54: StrokeDelta:      /      pressure:    -8
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 f7 80 ea 80 fc 00 00    00 00 28 0b 01 20 fe 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2043)
09:31:51 DEBUG: tuhi.protocol: offset 56: StrokeDelta:      /      pressure:    -9
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 ea 80 fc 00 00 00 00    28 0b 01 20 fe 28 02 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 58: StrokeDelta:      /      pressure:   -22
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 fc 00 00 00 00 28 0b    01 20 fe 28 02 01 28 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 60: StrokeDelta:      /      pressure:    -4
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 28 0b 01 20    fe 28 02 01 28 03 ff 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 62: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 28 0b 01 20 fe    28 02 01 28 03 ff 28 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 63: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 28 0b 01 20 fe 28    02 01 28 03 ff 28 03 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 64: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 28 0b 01 20 fe 28 02    01 28 03 ff 28 03 ff 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11241, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 65: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0b 01 20 fe 28 02 01    28 03 ff 28 03 ff 08 02 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11252, y=6082, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 66: StrokeDelta:   +11/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 fe 28 02 01 28 03 ff    28 03 ff 08 02 08 02 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11263, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 69: StrokeDelta:      /   -2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 02 01 28 03 ff 28 03    ff 08 02 08 02 08 02 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11276, y=6081, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 71: StrokeDelta:    +2/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 03 ff 28 03 ff 08 02    08 02 08 02 08 01 08 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11292, y=6080, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 74: StrokeDelta:    +3/   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 03 ff 08 02 08 02 08    02 08 01 08 01 20 ff 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11311, y=6078, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 77: StrokeDelta:    +3/   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 08 02 08 02 08 01    08 01 20 ff 28 01 01 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11332, y=6076, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 80: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 08 02 08 01 08 01    20 ff 28 01 01 28 ff fe …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11355, y=6074, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 82: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 08 01 08 01 20 ff    28 01 01 28 ff fe 28 fe …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11380, y=6072, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 84: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 01 08 01 20 ff 28 01    01 28 ff fe 28 fe 01 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11406, y=6070, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 86: StrokeDelta:    +1/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 01 20 ff 28 01 01 28    ff fe 28 fe 01 28 ff 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11433, y=6068, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 88: StrokeDelta:    +1/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 ff 28 01 01 28 ff fe    28 fe 01 28 ff 01 28 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11460, y=6065, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 90: StrokeDelta:      /   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 01 01 28 ff fe 28 fe    01 28 ff 01 28 ff ff 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11488, y=6063, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 92: StrokeDelta:    +1/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff fe 28 fe 01 28 ff    01 28 ff ff 28 ff 02 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11515, y=6059, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 95: StrokeDelta:    -1/   -2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 fe 01 28 ff 01 28 ff    ff 28 ff 02 08 fe 08 fd …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11540, y=6056, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 98: StrokeDelta:    -2/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff 01 28 ff ff 28 ff    02 08 fe 08 fd 28 fc 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11564, y=6054, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 101: StrokeDelta:    -1/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff ff 28 ff 02 08 fe    08 fd 28 fc 01 08 fd 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11587, y=6051, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 104: StrokeDelta:    -1/   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff 02 08 fe 08 fd 28    fc 01 08 fd 08 f6 08 09 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11609, y=6050, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 107: StrokeDelta:    -1/   +2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 fe 08 fd 28 fc 01 08    fd 08 f6 08 09 08 f7 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11629, y=6049, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 110: StrokeDelta:    -2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 fd 28 fc 01 08 fd 08    f6 08 09 08 f7 00 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11646, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 112: StrokeDelta:    -3/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 fc 01 08 fd 08 f6 08    09 08 f7 00 00 00 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11659, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 114: StrokeDelta:    -4/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 fd 08 f6 08 09 08 f7    00 00 00 00 00 28 f6 05 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11669, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 117: StrokeDelta:    -3/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 f6 08 09 08 f7 00 00    00 00 00 28 f6 05 28 0a …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11669, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 119: StrokeDelta:   -10/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 09 08 f7 00 00 00 00    00 28 f6 05 28 0a fb 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 121: StrokeDelta:    +9/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 f7 00 00 00 00 00 28    f6 05 28 0a fb 28 ec 02 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 123: StrokeDelta:    -9/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 00 28 f6 05    28 0a fb 28 ec 02 28 07 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 125: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 28 f6 05 28    0a fb 28 ec 02 28 07 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 126: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 28 f6 05 28 0a    fb 28 ec 02 28 07 ff 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 127: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 28 f6 05 28 0a fb    28 ec 02 28 07 ff 28 fb …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 128: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 28 f6 05 28 0a fb 28    ec 02 28 07 ff 28 fb 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11678, y=6048, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 129: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f6 05 28 0a fb 28 ec    02 28 07 ff 28 fb 01 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11668, y=6053, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 130: StrokeDelta:   -10/   +5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0a fb 28 ec 02 28 07    ff 28 fb 01 08 ff 28 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11668, y=6053, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 133: StrokeDelta:   +10/   -5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ec 02 28 07 ff 28 fb    01 08 ff 28 ff ff 08 02 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11648, y=6055, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 136: StrokeDelta:   -20/   +2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 07 ff 28 fb 01 08 ff    28 ff ff 08 02 08 02 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11635, y=6056, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 139: StrokeDelta:    +7/   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 fb 01 08 ff 28 ff ff    08 02 08 02 08 02 00 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11617, y=6058, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 142: StrokeDelta:    -5/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 ff 28 ff ff 08 02 08    02 08 02 00 28 01 01 20 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11598, y=6060, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 145: StrokeDelta:    -1/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff ff 08 02 08 02 08    02 00 28 01 01 20 fd 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11578, y=6061, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 147: StrokeDelta:    -1/   -1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 08 02 08 02 00 28    01 01 20 fd 28 01 01 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11560, y=6062, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 150: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 08 02 00 28 01 01    20 fd 28 01 01 28 03 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11544, y=6063, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 152: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 02 00 28 01 01 20 fd    28 01 01 28 03 01 20 fe …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11530, y=6064, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 154: StrokeDelta:    +2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 28 01 01 20 fd 28 01    01 28 03 01 20 fe 20 01 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11516, y=6065, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 156: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 01 01 20 fd 28 01 01    28 03 01 20 fe 20 01 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11503, y=6067, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 157: StrokeDelta:    +1/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 fd 28 01 01 28 03 01    20 fe 20 01 08 09 28 f1 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11490, y=6066, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 160: StrokeDelta:      /   -3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 01 01 28 03 01 20 fe    20 01 08 09 28 f1 fe 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11478, y=6066, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 162: StrokeDelta:    +1/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 03 01 20 fe 20 01 08    09 28 f1 fe 28 0f 02 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11469, y=6067, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 165: StrokeDelta:    +3/   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 fe 20 01 08 09 28 f1    fe 28 0f 02 28 f7 fd 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11460, y=6066, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 168: StrokeDelta:      /   -2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 01 08 09 28 f1 fe 28    0f 02 28 f7 fd 28 09 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11451, y=6066, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 170: StrokeDelta:      /   +1 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 09 28 f1 fe 28 0f 02    28 f7 fd 28 09 03 00 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11451, y=6066, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 172: StrokeDelta:    +9/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f1 fe 28 0f 02 28 f7    fd 28 09 03 00 28 f6 fb …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11436, y=6064, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 174: StrokeDelta:   -15/   -2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0f 02 28 f7 fd 28 09    03 00 28 f6 fb 28 0a 05 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11436, y=6064, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 177: StrokeDelta:   +15/   +2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f7 fd 28 09 03 00 28    f6 fb 28 0a 05 00 28 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11427, y=6061, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 180: StrokeDelta:    -9/   -3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 09 03 00 28 f6 fb 28    0a 05 00 28 ff f2 28 ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11427, y=6061, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 183: StrokeDelta:    +9/   +3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 28 f6 fb 28 0a 05 00    28 ff f2 28 ff 02 28 fc …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11427, y=6061, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 186: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f6 fb 28 0a 05 00 28    ff f2 28 ff 02 28 fc fb …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11417, y=6056, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 187: StrokeDelta:   -10/   -5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0a 05 00 28 ff f2 28    ff 02 28 fc fb 28 fe fc …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11417, y=6056, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 190: StrokeDelta:   +10/   +5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 28 ff f2 28 ff 02 28    fc fb 28 fe fc 28 f9 fd …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11417, y=6056, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 193: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff f2 28 ff 02 28 fc    fb 28 fe fc 28 f9 fd 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11416, y=6042, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 194: StrokeDelta:    -1/  -14 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff 02 28 fc fb 28 fe    fc 28 f9 fd 28 ff fd 08 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11414, y=6030, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 197: StrokeDelta:    -1/   +2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 fc fb 28 fe fc 28 f9    fd 28 ff fd 08 fe 28 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11408, y=6013, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 200: StrokeDelta:    -4/   -5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 fe fc 28 f9 fd 28 ff    fd 08 fe 28 03 06 28 07 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11400, y=5992, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 203: StrokeDelta:    -2/   -4 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f9 fd 28 ff fd 08 fe    28 03 06 28 07 02 28 05 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11385, y=5968, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 206: StrokeDelta:    -7/   -3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 ff fd 08 fe 28 03 06    28 07 02 28 05 08 28 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11369, y=5941, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 209: StrokeDelta:    -1/   -3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 08 fe 28 03 06 28 07 02    28 05 08 28 03 0b 28 0b …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11351, y=5914, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 212: StrokeDelta:    -2/      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 03 06 28 07 02 28 05    08 28 03 0b 28 0b f8 20 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11336, y=5893, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 214: StrokeDelta:    +3/   +6 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 07 02 28 05 08 28 03    0b 28 0b f8 20 05 28 f5 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11328, y=5874, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 217: StrokeDelta:    +7/   +2 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 05 08 28 03 0b 28 0b    f8 20 05 28 f5 03 28 0e …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11325, y=5863, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 220: StrokeDelta:    +5/   +8 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 03 0b 28 0b f8 20 05    28 f5 03 28 0e f6 28 f2 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11325, y=5863, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 223: StrokeDelta:    +3/  +11 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0b f8 20 05 28 f5 03    28 0e f6 28 f2 0a 28 f6 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11336, y=5855, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 226: StrokeDelta:   +11/   -8 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 20 05 28 f5 03 28 0e f6    28 f2 0a 28 f6 f1 28 0a …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11347, y=5852, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 229: StrokeDelta:      /   +5 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f5 03 28 0e f6 28 f2    0a 28 f6 f1 28 0a 0f 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11347, y=5852, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 231: StrokeDelta:   -11/   +3 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0e f6 28 f2 0a 28 f6    f1 28 0a 0f 28 f3 f8 28 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11361, y=5842, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 234: StrokeDelta:   +14/  -10 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f2 0a 28 f6 f1 28 0a    0f 28 f3 f8 28 0d 08 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11361, y=5842, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 237: StrokeDelta:   -14/  +10 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f6 f1 28 0a 0f 28 f3    f8 28 0d 08 00 00 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11351, y=5827, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 240: StrokeDelta:   -10/  -15 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0a 0f 28 f3 f8 28 0d    08 00 00 00 00 00 00 00 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11351, y=5827, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 243: StrokeDelta:   +10/  +15 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 f3 f8 28 0d 08 00 00    00 00 00 00 00 80 fe 80 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 246: StrokeDelta:   -13/   -8 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 28 0d 08 00 00 00 00 00    00 00 80 fe 80 80 80 91 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 249: StrokeDelta:   +13/   +8 pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 00 00 00 80    fe 80 80 80 91 c0 14 05 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 252: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 00 00 80 fe    80 80 80 91 c0 14 05 e8 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 253: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 00 80 fe 80    80 80 91 c0 14 05 e8 0c …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 254: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 00 80 fe 80 80    80 91 c0 14 05 e8 0c 11 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 255: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 00 80 fe 80 80 80    91 c0 14 05 e8 0c 11 99 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 256: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 00 80 fe 80 80 80 91    c0 14 05 e8 0c 11 99 03 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 257: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 00 80 fe 80 80 80 91 c0    14 05 e8 0c 11 99 03 e8 …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2047)
09:31:51 DEBUG: tuhi.protocol: offset 258: StrokeDelta:      /      pressure:      
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 fe 80 80 80 91 c0 14    05 e8 0c 11 99 03 e8 fe …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=2045)
09:31:51 DEBUG: tuhi.protocol: offset 259: StrokeDelta:      /      pressure:    -2
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 80 80 91 c0 14 05 e8    0c 11 99 03 e8 fe f9 3c …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=1915)
09:31:51 DEBUG: tuhi.protocol: offset 261: StrokeDelta:      /      pressure:  -128
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: 80 91 c0 14 05 e8 0c 11    99 03 e8 fe f9 3c 01 fc …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=1674)
09:31:51 DEBUG: tuhi.protocol: offset 263: StrokeDelta:      /      pressure:  -111
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: c0 14 05 e8 0c 11 99 03    e8 fe f9 3c 01 fc ff ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11338, y=5819, p=1300)
09:31:51 DEBUG: tuhi.protocol: offset 265: StrokeDelta:      /      pressure:  1300
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 0c 11 99 03 e8 fe f9    3c 01 fc ff ff ff ff ff …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11350, y=5836, p=921)
09:31:51 DEBUG: tuhi.protocol: offset 268: StrokeDelta:   +12/  +17 pressure:   921
09:31:51 DEBUG: tuhi.protocol: next data packet DELTA: e8 fe f9 3c 01 fc ff ff    ff ff ff ff ff ff ee ee …
09:31:51 DEBUG: tuhi.protocol: Calculated point: Point(x=11360, y=5846, p=316)
09:31:51 DEBUG: tuhi.protocol: offset 273: StrokeDelta:    -2/   -7 pressure:   316
09:31:51 DEBUG: tuhi.protocol: next data packet EOF: fc ff ff ff ff ff ff ff    ff ee ee ee 6d b6 00 00 …
09:31:51 DEBUG: tuhi: Drawing received
09:31:51 DEBUG: tuhi.config: E0:61:C6:BF:14:4E: adding new drawing, timestamp 1566286302
09:31:51 DEBUG: tuhi.fw: command: ack transaction
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> ca / 01 / 00
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- b3 01 00
09:31:51 DEBUG: tuhi.fw: command: check for data
09:31:51 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: TX NORDIC --> c1 / 01 / 00
09:31:52 DEBUG: tuhi.fw: E0:61:C6:BF:14:4E: RX NORDIC <-- c2 02 00 00
09:31:52 DEBUG: tuhi.wacom: Drawings available: 0
09:31:52 INFO: tuhi.ble.E0:61:C6:BF:14:4E: Disconnecting
09:31:54 INFO: tuhi.ble.E0:61:C6:BF:14:4E: Disconnected
09:31:54 DEBUG: tuhi: E0:61:C6:BF:14:4E: disconnected
```

Here is the full session log with a bigger drawing where only the first stroke gets recognized: 
[tuhi-19-08-20-09:27:35.log](https://github.com/tuhiproject/tuhi/files/3519223/tuhi-19-08-20-09.27.35.log)

Note that the pen data gets correctly retrieved.

bentiss commented:
Here are the raw logs, they seem rather short: 
[logs.txt](https://github.com/tuhiproject/tuhi/files/3519599/logs.txt)

(got to rename the `.yaml` into `.txt`)

bentiss commented:
I think I got the correct one here: 
[log-1566286650-2019-08-20-09:37:30.txt](https://github.com/tuhiproject/tuhi/files/3519642/log-1566286650-2019-08-20-09.37.30.txt)

whot committed:
wacom: store the default orientation for each tablet

Not actually used at this point, but this way we have default
width/height/orientation etc. all in one place.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

whot commented:
Fixed, I had some wrong parsing in the stroke end vs end of file packets.

The svg test tool now takes a `--tablet-mode=slate` argument to use the defaults for the given tablet.

bentiss commented:
works now, thanks!

bentiss reviewed (approved):
LGTM
